shell/platform/darwin/ios/framework/Source/FlutterTextInputPlugin.mm

```diff
@@ -2454,7 +2394,6 @@ - (void)cleanUpViewHierarchy:(BOOL)includeActiveView
         if (clearText) {
           [inputView replaceRangeLocal:NSMakeRange(0, inputView.text.length) withText:@""];
         }
-        [inputView decommission];
```

Comment:
Do we need to nil out the delegate here? The original code is strange because the `_decommissioned` ivar doesn't seem to be used anywhere except for an `NSAssert`, but it could be a bug)

Reply:
I think the original issue was that a `FlutterInputView` can be retained by UIKit even if it's no longer in the view hierarchy, so it can outlive the engine instance, and when that happens the reference to the engine becomes a dangling pointer. With ARC I think `textInputDelegate` will be set to `nil` once the engine is deallocated.